marshal/admin.go

```diff
@@ -303,14 +330,22 @@ func (a *consumerGroupAdmin) SetConsumerGroupPosition(groupID string,
 	defer close(claimFailures)
 
 	var claimsWg sync.WaitGroup
-	// stopHeartbeating channel instructs all successfully claimed and heartbeating claims to stop.
-	stopHeartbeating := make(chan struct{})
+
+	// Closing stopHeartbeats instructs all successfully claimed and heartbeating claims to stop.
+	// Waiting on heartbeatsWg will ensure that all heartbeating has in fact stopped.  Note that
+	// it's important to define heartbeatsWg as a reference so that method calls with heartbeatsWg
+	// as a parameter actually use the same WaitGroup (instead of calling Done on a copy, which
+	// will make the below calls to Wait block forever).
```

Comment:
Randomly, it's Jetstream policy to always use `sync` structs (Mutex, RWMutex, WaitGroup, etc) as reference. We don't allow them to be values. This has saved us several times. :-)